services/src/main/scala/cromwell/services/loadcontroller/impl/LoadControllerServiceActor.scala

```diff
@@ -0,0 +1,69 @@
+package cromwell.services.loadcontroller.impl
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Timers}
+import akka.routing.Listeners
+import cats.data.NonEmptyList
+import com.typesafe.config.Config
+import cromwell.core.actor.BatchActor.QueueWeight
+import cromwell.core.instrumentation.InstrumentationPrefixes
+import cromwell.services.instrumentation.CromwellInstrumentation
+import cromwell.services.loadcontroller.LoadControllerService._
+import cromwell.services.loadcontroller.LoadMetric
+import cromwell.services.loadcontroller.impl.LoadControllerServiceActor._
+import cromwell.services.metadata.MetadataService.ListenToMetadataWriteActor
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.duration._
+
+object LoadControllerServiceActor {
+  val LoadControllerServiceName = "LoadController"
+  val LoadLevelInstrumentation = NonEmptyList.one("loadLevel")
+  case object LoadControlTimerKey
+  case object LoadControlTimerAction
+}
+
+/**
+  * Service Actor that monitors load and sends alert to the rest of the system when load is determined abnormal.
+  */
+class LoadControllerServiceActor(serviceConfig: Config, globalConfig: Config, override val serviceRegistryActor: ActorRef) extends Actor
+  with ActorLogging with Listeners with Timers with CromwellInstrumentation {
+  private val controlFrequency = serviceConfig.as[Option[FiniteDuration]]("control-frequency").getOrElse(5.seconds)
+  private val metadataQueueThreshold = serviceConfig.as[Option[Int]]("metadata-queue-threshold").getOrElse(1 * 1000 * 1000)
+  private val metadataQueueMetric = MetadataQueueMetric(metadataQueueThreshold)
+
+  private var loadLevel: LoadLevel = NormalLoad
+  private var loadMetrics: Map[LoadMetric, LoadLevel] = Map.empty
+
+  override def receive = listenerManagement.orElse(controlReceive)
+
+  override def preStart() = {
+    serviceRegistryActor ! ListenToMetadataWriteActor
```

Comment:
I think I'm going to change this from "this actor needs to ask the right actors to send it updates" to "whoever wants can send `LoadMetric`s to this actor and it will aggregate them and take action if necessary"

Reply:
Is something guaranteed to notice if one of those "right actors" is AWOL and therefore its `LoadMetric`s never arrive?

Reply:
Good question, presumably if one of the actors goes AWOL it's going to be outside the scope of what this actor can do to fix it (at least for now).
But we could have it actor could watch every new actor that starts sending metrics to it and if one of them dies it could automatically raise the load level ?